src/resource.h

```diff
@@ -94,6 +95,8 @@ class Resource {
   /// Returns the package id.
   virtual std::string package_name() { return Package::unpackaged_name(); };
 
+  virtual Ptr PackageExtract(double qty, std::string new_package_name = Package::unpackaged_name()) = 0;
```

Comment:
I'm not sure you need to define a default here since this is a pure virtual function
```suggestion
  virtual Ptr PackageExtract(double qty, std::string new_package_name) = 0;
```